frontend/src/actions/ilabActions.js

```diff
@@ -556,23 +555,15 @@ export const checkIlabJobs = (newPage) => (dispatch, getState) => {
 };
 
 /**
- * Add a new metric to the selected list if not present, or remove it if it
+ * Updated the selected metric list
  * was previously present.
  *
  * @param {string} metric
  */
-export const toggleSelectedMetric = (metric) => (dispatch, getState) => {
-  let metrics_selected = getState().ilab.metrics_selected;
-  if (metrics_selected.includes(metric)) {
-    metrics_selected = metrics_selected.filter((m) => m !== metric);
-  } else {
-    metrics_selected = [...metrics_selected, metric];
-  }
-  dispatch({
-    type: TYPES.SET_ILAB_SELECTED_METRICS,
-    payload: metrics_selected,
-  });
-};
+export const toggleSelectedMetric = (metric) => ({
+  type: TYPES.SET_ILAB_SELECTED_METRICS,
+  payload: metric,
+});
```

Comment:
`SET_ILAB_SELECTED_METRICS` stores the *set* of selected metrics; whereas `toggleSelectedMetric(metric)` used to flip the state of the single named metric and then store the updated set, this modified implementation just passes through the set. Therefore, both the name (it doesn't "toggle") and the parameter (it's no longer *a* metric) should be changed to match the new implementation.